### PR TITLE
Update Error and OpCodes to match docs

### PR DESCRIPTION
**Pull request by DRSchlaubi**

### CODE REVIEW

**Review by HopeBaron — APPROVED** [review]

**Comment by HopeBaron** [comment]
looks fine, although needs fixing before being merged.

**Comment by DRSchlaubi** [comment]
> looks fine, although needs fixing before being merged.

Did you notice how the commit which fixed it made no changes?

File: rest/src/main/kotlin/json/JsonErrorCode.kt [hunk 1]
```diff
@@ -179,20 +193,48 @@ enum class JsonErrorCode(val code: Int) {
...
+
     /**
      * Unknown guild template.
      *
      * [JSON Error Codes](https://github.com/discord/discord-api-docs/blob/master/docs/topics/Opcodes_and_Status_Codes.md#json-error-codes)
      */
     UnknownGuildTemplate(10057),
 
+    /**
+     * Unknown discoverable server category-
```
Comment (BartArys) [inline rest/src/main/kotlin/json/JsonErrorCode.kt:225]:
```suggestion
     * Unknown discoverable server category.
```

File: rest/src/main/kotlin/json/JsonErrorCode.kt [hunk 2]
```diff
@@ -452,13 +676,48 @@ enum class JsonErrorCode(val code: Int) {
...
+
+    /**
+     * Invalid webhook token provided.
+     *
+     * [JSON Error Codes](https://github.com/discord/discord-api-docs/blob/master/docs/topics/Opcodes_and_Status_Codes.md#json-error-codes)
+     */
+    InvalidWebhookToken(50027),
+
+    /**
+     * Invalid role,
```
Comment (BartArys) [inline rest/src/main/kotlin/json/JsonErrorCode.kt:708]:
```suggestion
     * Invalid role.
```

File: rest/src/main/kotlin/json/JsonErrorCode.kt [hunk 3]
```diff
@@ -452,13 +676,48 @@ enum class JsonErrorCode(val code: Int) {
...
+     *
+     * [JSON Error Codes](https://github.com/discord/discord-api-docs/blob/master/docs/topics/Opcodes_and_Status_Codes.md#json-error-codes)
+     */
+    InvalidRole(50028),
+
+    /**
+     * "Invalid Recipient(s)"
```
Comment (BartArys) [inline rest/src/main/kotlin/json/JsonErrorCode.kt:715]:
```suggestion
     * Invalid Recipient(s).
```

File: rest/src/main/kotlin/json/JsonErrorCode.kt [hunk 4]
```diff
@@ -502,6 +775,20 @@ enum class JsonErrorCode(val code: Int) {
      */
     InvalidStickerSent(50081),
 
+    /**
+     *	Invalid thread notification settings.
```
Comment (BartArys) [inline rest/src/main/kotlin/json/JsonErrorCode.kt:779]:
```suggestion
     * Invalid thread notification settings.
```

**Review by BartArys — APPROVED** [review]

**Pull request merged**